### PR TITLE
Move Sequence to Task for RecoTauTag RecoCTPPS

### PR DESCRIPTION
**Pull request by jeongeun**
PR description: 
Move Sequence to Task for **RecoTauTag  RecoCTPPS** cff python files. 
(An extension of the previous task [PR#25163](https://github.com/cms-sw/cmssw/pull/25163), [PR#27474](https://github.com/cms-sw/cmssw/pull/27474) , [PR#27640](https://github.com/cms-sw/cmssw/pull/27640), [PR#27887](https://github.com/cms-sw/cmssw/pull/27887), [PR#28050](https://github.com/cms-sw/cmssw/pull/28050), [PR#28300](https://github.com/cms-sw/cmssw/pull/28300) )

Tested in CMSSW_11_0_X_2019_11-12-2300, the test all passed in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html).

### CODE REVIEW

**Comment by perrotta** [comment]
please test

File: RecoEgamma/EgammaPhotonProducers/python/conversionTrackSequenceForReReco_cff.py [hunk 1]
```diff
@@ -8,7 +8,10 @@
 # Conversion Track producer  ( final fit )
 from RecoEgamma.EgammaPhotonProducers.ckfOutInTracksFromConversions_cfi import *
 from RecoEgamma.EgammaPhotonProducers.ckfInOutTracksFromConversions_cfi import *
-ckfTracksFromConversionsReReco = cms.Sequence(conversionTrackCandidates*ckfOutInTracksFromConversions*ckfInOutTracksFromConversions)
+ckfTracksFromConversionsReRecoTask = cms.Task(conversionTrackCandidates,
+                                              ckfOutInTracksFromConversions,
+                                              ckfInOutTracksFromConversions)
+ckfTracksFromConversionsReReco = cms.Sequence(ckfTracksFromConversionsReRecoTask)
```
Comment (perrotta) [inline RecoEgamma/EgammaPhotonProducers/python/conversionTrackSequenceForReReco_cff.py]: This is only used here internally: no need to define the sequence

File: RecoEgamma/EgammaPhotonProducers/python/conversionTrackSequenceForReReco_cff.py [hunk 2]
```diff
@@ -46,7 +49,12 @@
     useTrajectory = cms.bool(False),
 )
 
-conversionTrackProducersReReco = cms.Sequence(generalConversionTrackProducerReReco*conversionStepConversionTrackProducerReReco*inOutConversionTrackProducerReReco*outInConversionTrackProducerReReco*gsfConversionTrackProducerReReco)
+conversionTrackProducersReRecoTask = cms.Task(generalConversionTrackProducerReReco,
+                                              conversionStepConversionTrackProducerReReco,
+                                              inOutConversionTrackProducerReReco,
+                                              outInConversionTrackProducerReReco,
+                                              gsfConversionTrackProducerReReco)
+conversionTrackProducersReReco = cms.Sequence(conversionTrackProducersReRecoTask)
```
Comment (perrotta) [inline RecoEgamma/EgammaPhotonProducers/python/conversionTrackSequenceForReReco_cff.py]: This is only used here internally: no need to define the sequence

File: RecoEgamma/EgammaPhotonProducers/python/conversionTrackSequenceForReReco_cff.py [hunk 3]
```diff
@@ -101,9 +109,16 @@
 #overlaps between the ecal seeded track collections and between ecal seeded and general tracks are arbitrated first by nhits then by chi^2/dof
 #(logic and much of the code is adapted from FinalTrackSelectors)
 
-conversionTrackMergersReReco = cms.Sequence(inOutOutInConversionTrackMergerReReco*generalConversionStepConversionTrackMergerReReco*generalInOutOutInConversionTrackMergerReReco*gsfGeneralInOutOutInConversionTrackMergerReReco)
+conversionTrackMergersReRecoTask = cms.Task(inOutOutInConversionTrackMergerReReco,
+                                            generalConversionStepConversionTrackMergerReReco,
+                                            generalInOutOutInConversionTrackMergerReReco,
+                                            gsfGeneralInOutOutInConversionTrackMergerReReco)
+conversionTrackMergersReReco = cms.Sequence(conversionTrackMergersReRecoTask)
```
Comment (perrotta) [inline RecoEgamma/EgammaPhotonProducers/python/conversionTrackSequenceForReReco_cff.py]: This is only used here internally: no need to define the sequence

File: RecoEgamma/EgammaIsolationAlgos/python/eleIsoDeposits_cff.py [hunk 4]
```diff
@@ -6,11 +6,11 @@
...
-    eleIsoDepositHcalDepth1FromTowers +
+eleIsoDepositsTask = cms.Task(
+    eleIsoDepositTk ,
+    eleIsoDepositEcalFromHits , 
+    eleIsoDepositHcalFromTowers ,
+    eleIsoDepositHcalDepth1FromTowers ,
     eleIsoDepositHcalDepth2FromTowers
 )
-
+eleIsoDeposits = cms.Sequence(eleIsoDepositsTask)
```
Comment (perrotta) [inline RecoEgamma/EgammaIsolationAlgos/python/eleIsoDeposits_cff.py]: This is only used in ```RecoEgamma/EgammaIsolationAlgos/python/eleIsolationSequence_cff.py```: maybe one could avoid defining explicitly the sequence here

File: RecoEgamma/EgammaIsolationAlgos/python/eleIsoFromDeposits_cff.py [hunk 5]
```diff
@@ -2,11 +2,11 @@
...
-    eleIsoFromDepsHcalDepth1FromTowers *
+eleIsoFromDepositsTask = cms.Task( 
+    eleIsoFromDepsTk , 
+    eleIsoFromDepsEcalFromHitsByCrystal , 
+    eleIsoFromDepsHcalFromTowers ,
+    eleIsoFromDepsHcalDepth1FromTowers ,
     eleIsoFromDepsHcalDepth2FromTowers
 )
-
+eleIsoFromDeposits = cms.Task(eleIsoFromDepositsTask)
```
Comment (perrotta) [inline RecoEgamma/EgammaIsolationAlgos/python/eleIsoFromDeposits_cff.py]: This is only used in ```RecoEgamma/EgammaIsolationAlgos/python/eleIsolationSequence_cff.py```: maybe one could avoid defining explicitly the sequence here

File: RecoEgamma/EgammaIsolationAlgos/python/gamIsoDeposits_cff.py [hunk 6]
```diff
@@ -6,11 +6,11 @@
...
-    gamIsoDepositHcalDepth1FromTowers +
+gamIsoDepositsTask = cms.Task(
+    gamIsoDepositTk , 
+    gamIsoDepositEcalFromHits , 
+    gamIsoDepositHcalFromTowers ,
+    gamIsoDepositHcalDepth1FromTowers ,
     gamIsoDepositHcalDepth2FromTowers
 )
-
+gamIsoDeposits = cms.Sequence(gamIsoDepositsTask)
```
Comment (perrotta) [inline RecoEgamma/EgammaIsolationAlgos/python/gamIsoDeposits_cff.py]: This is only used in ```RecoEgamma/EgammaIsolationAlgos/python/gamIsolationSequence_cff.py```: maybe one could avoid defining explicitly the sequence here

File: RecoEgamma/EgammaIsolationAlgos/python/gamIsoFromDeposits_cff.py [hunk 7]
```diff
@@ -2,11 +2,11 @@
...
-    gamIsoFromDepsHcalDepth1FromTowers *
+gamIsoFromDepositsTask = cms.Task( 
+    gamIsoFromDepsTk ,
+    gamIsoFromDepsEcalFromHitsByCrystal , 
+    gamIsoFromDepsHcalFromTowers ,
+    gamIsoFromDepsHcalDepth1FromTowers ,
     gamIsoFromDepsHcalDepth2FromTowers
 )
-
+gamIsoFromDeposits = cms.Sequence(gamIsoFromDepositsTask)
```
Comment (perrotta) [inline RecoEgamma/EgammaIsolationAlgos/python/gamIsoFromDeposits_cff.py]: This is only used in ```RecoEgamma/EgammaIsolationAlgos/python/gamIsolationSequence_cff.py```: maybe one could avoid defining explicitly the sequence here

File: RecoEgamma/EgammaIsolationAlgos/python/gamIsolationSequence_cff.py [hunk 8]
```diff
@@ -8,4 +8,3 @@
     gamIsoDepositsTask,  
     gamIsoFromDepositsTask
 )
-gamIsolationSequence = cms.Sequence(gamIsolationTask)
```
Comment (perrotta) [inline RecoEgamma/EgammaIsolationAlgos/python/gamIsolationSequence_cff.py]: Please don't remove this one: it is still used in RecoEgamma/EgammaIsolationAlgos/python/egammaIsolationSequencePAT_cff.py

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- The implementation of Tasks to replace what previously included only as sequences in the affected packages is made similarly to what already implemented for other packages
- Jenkins tests pass and show no differences

**Comment by fabiocos** [comment]
+1

**Pull request merged**